src/agent/src/watcher.rs

```diff
@@ -258,18 +261,22 @@ impl SandboxStorages {
                 continue;
             }
 
-            match entry.scan(logger).await {
-                Err(error) => {
+            entry
+                .clone()
+                .scan(logger)
+                .or_else(|e| async move {
                     // If an error was observed, we will stop treating this Storage as being watchable, and
                     // instead clean up the target-mount files on the tmpfs and bind mount the source_mount_point
                     // to target_mount_point.
-                    error!(logger, "error observed when watching: {:?}", error);
+                    error!(logger, "error observed when watching: {:?}", e);
                     entry.watch = false;
 
                     // Remove destination contents, but not the directory itself, since this is
                     // assumed to be bind-mounted into a container. If source/mount is a file, no need to cleanup
                     if entry.target_mount_point.as_path().is_dir() {
-                        for dir_entry in std::fs::read_dir(entry.target_mount_point.as_path())? {
+                        for dir_entry in std::fs::read_dir(entry.target_mount_point.as_path())
```

Comment:
What's this giving me, versus just `?`

Reply:
(to clarify, I just don't understand; not necessarily questioning why we should!)

Reply:
You're right, this is pointless, fixed now.